code/modules/mob/living/slime/_slime.dm

```diff
@@ -0,0 +1,333 @@
+		PRINT_STACK_TRACE("Slime initialized with non-decl slime type: [slime_type || "NULL"]")
+		return INITIALIZE_HINT_QDEL
+
+	number = random_id(/mob/living/slime, 1, 1000)
+	mutation_chance = rand(25, 35)
+	update_name()
+	regenerate_icons()
+
+/mob/living/slime/proc/update_name()
+	var/decl/slime_colour/slime_data = decls_repository.get_decl(slime_type)
+	SetName("[slime_data.name] [is_adult ? "adult" : "baby"] slime ([number])")
+
+/mob/living/slime/movement_delay()
+	if (bodytemperature >= 330.23) // 135 F
+		return -1	// slimes become supercharged at high temperatures
+
+	var/tally = ..()
+
+	var/health_deficiency = (maxHealth - health)
+	if(health_deficiency >= 30) tally += (health_deficiency / 25)
+
+	if (bodytemperature < 183.222)
+		tally += (283.222 - bodytemperature) / 10 * 1.75
+
+	if(reagents)
+		if(reagents.has_reagent(/decl/material/liquid/amphetamines)) // Stimulants slows slimes down
+			tally *= 2
+
+		if(reagents.has_reagent(/decl/material/liquid/frostoil)) // Frostoil also makes them move VEEERRYYYYY slow
+			tally *= 5
+
+	if(health <= 0) // if damaged, the slime moves twice as slow
+		tally *= 2
+
+	return tally + config.slime_delay
+
+/mob/living/slime/Bump(atom/movable/AM, yes)
+	if ((!(yes) || now_pushing))
+		return
+	now_pushing = 1
+
+	if(isobj(AM) && !client && powerlevel > 0)
+		var/probab = 10
+		switch(powerlevel)
+			if(1 to 2)	probab = 20
+			if(3 to 4)	probab = 30
+			if(5 to 6)	probab = 40
+			if(7 to 8)	probab = 60
+			if(9)		probab = 70
+			if(10)		probab = 95
+		if(prob(probab))
+			if(istype(AM, /obj/structure/window) || istype(AM, /obj/structure/grille))
+				if(nutrition <= get_hunger_nutrition())
+					if (is_adult || prob(5))
+						UnarmedAttack(AM)
+
+	if(ismob(AM))
+		var/mob/tmob = AM
+
+		if(is_adult)
+			if(istype(tmob, /mob/living/carbon/human))
+				if(prob(90))
+					now_pushing = 0
+					return
+		else
+			if(istype(tmob, /mob/living/carbon/human))
+				now_pushing = 0
+				return
+
+	now_pushing = 0
+
+	..()
+
+/mob/living/slime/Stat()
+	. = ..()
+
+	statpanel("Status")
+	stat(null, "Health: [round((health / maxHealth) * 100)]%")
+	stat(null, "Intent: [a_intent]")
+
+	if (client.statpanel == "Status")
+		stat(null, "Nutrition: [nutrition]/[get_max_nutrition()]")
+		if(amount_grown >= 10)
+			if(is_adult)
+				stat(null, "You can reproduce!")
+			else
+				stat(null, "You can evolve!")
+
+		stat(null,"Power Level: [powerlevel]")
+
+/mob/living/slime/adjustFireLoss(amount)
+	..(-abs(amount)) // Heals them
+	return
+
+/mob/living/slime/bullet_act(var/obj/item/projectile/Proj)
+	var/datum/ai/slime/slime_ai = ai
+	if(istype(slime_ai))
+		slime_ai.attacked += 10
+		slime_ai.adjust_friendship(Proj.firer, -5)
+	..(Proj)
+	return 0
+
+/mob/living/slime/emp_act(severity)
+	powerlevel = 0 // oh no, the power!
+	..()
+
+/mob/living/slime/explosion_act(severity)
+	..()
+	switch(severity)
+		if(1)
+			qdel(src)
+		if(2)
+			adjustBruteLoss(60)
+			adjustFireLoss(60)
```

Comment:
Is it intentional that this heals them?

Reply:
Going to revisit this chunk of code, might rip it out entirely.